movement/watch_faces/complication/stock_stopwatch_face.c

```diff
@@ -0,0 +1,246 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Andreas Nebinger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include "stock_stopwatch_face.h"
+#include "watch.h"
+#include "watch_utility.h"
+#include "watch_rtc.h"
+
+/*
+    This watch face implements the original F-91W stopwatch functionality
+    including counting hundredths of seconds and lap timing. There are two
+    improvements compared to the functionality of the original: 
+    1. When reaching 59:59 the counter does not simply jump back to zero,
+       but keeps track of hours in the upper right hand corner. (Up to 24h)
+    2. Long pressing the light button toggles the led behaviour: it either
+       turns on on each button press or it doesn't.
+*/
+
+
+static int8_t _callback_slot;
+static uint32_t _ticks;
+static uint32_t _lap_ticks;
+static uint8_t _blink_ticks;
+static uint32_t _old_seconds;
+static uint8_t _old_minutes;
+static uint8_t _hours;
+static bool _colon;
+static bool _is_running;
+static bool _light_on_button;
+
+static void _fast_tick_callback() {
+    _ticks++;
+    // overflow is handled further down in the app loop, to keep this callback is as simple as possible
+}
+
+static inline void _button_beep(movement_settings_t *settings) {
+    // play a beep as confirmation for a button press (if applicable)
+    if (settings->bit.button_should_sound) watch_buzzer_play_note(BUZZER_NOTE_C7, 50);
+}
+
+static void _display_ticks(uint32_t ticks) {
+    // show minutes, seconds and fractions derived from 128 Hz tick counter
+    char buf[14];
+    uint8_t sec_100 = (ticks & 0x7F) * 100 / 128;
+    uint32_t seconds = ticks >> 7;
+    uint32_t minutes = seconds / 60;
+    if (_hours)
+        sprintf(buf, "%2u%02lu%02lu%02u", _hours, minutes, (seconds % 60), sec_100);
+    else
+        sprintf(buf, "  %02lu%02lu%02u", minutes, (seconds % 60), sec_100);
```

Comment:
The clock face does a bit of work to figure out what updates are necessary; this is likely to pay off more here, I suspect.